### DISCUSSION TITLE
How to install a persistent NetBox plug-in?

### DISCUSSION THREAD

**Discussion by LWJ-SARA** [cisagov/Malcolm#329] [Q&A] [answered]
Hello, I am looking to integrate the Netbox-Topology-Views plugin into the deployed malcolm component netbox (v3.5.7). However, upon restarting the pod post-installation, all changes are lost. What steps should I take to ensure persistent installation? Could you kindly provide a detailed list of instructions? Thank you.

**Comment by mmguero** [comment]
Hi! This is something we're tracking [as a feature request](https://github.com/idaholab/Malcolm/issues/530) for an upcoming version. Hopefully until that's complete the details listed in the issue I've linked there can answer your question.

**Comment by mmguero** [comment, marked as answer]
In the next release this will be handled. From the new documentation that will accompany that release:

## NetBox Plugins

NetBox's functionality can be extended with plugins that can provide "[new data models, integrations, and more](https://netboxlabs.com/netbox-plugins/)" (see also the [NetBox Wiki](https://github.com/netbox-community/netbox/wiki/Plugins)).

When Malcolm's NetBox container [starts up](https://github.com/mmguero-dev/Malcolm/blob/development/netbox/scripts/netbox_install_plugins.py), it installs (using [pip](https://packaging.python.org/en/latest/guides/tool-recommendations/#installing-packages)) any NetBox plugins that have [cloned](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository) or [downloaded and extracted](https://docs.github.com/en/repositories/working-with-files/using-files/downloading-source-code-archives) into subdirectories in `./netbox/custom-plugins/` in the Malcolm installation directory. In instances where Malcolm is being run in an offline/airgapped configuration, the plugins' additional dependencies must also be present under `./netbox/custom-plugins/requirements/`, where they will be automatically installed first.

The following warning is quoted from the [NetBox documentation](https://netboxlabs.com/docs/netbox/en/stable/configuration/plugins/):

> Plugins extend NetBox by allowing external code to run with the same access and privileges as NetBox itself. Only install plugins from trusted sources. The NetBox maintainers make absolutely no guarantees about the integrity or security of your installation with plugins enabled.
